libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/utils/RouterEx.kt

```diff
@@ -0,0 +1,9 @@
+package com.mapbox.navigation.core.internal.utils
+
+import com.mapbox.navigation.base.internal.route.InternalRouter
+import com.mapbox.navigation.base.route.Router
+
+/**
+ * Check if [Router] interface impl in Nav SDK
+ */
+internal fun Router.isInternalImplementation(): Boolean = this is InternalRouter
```

Comment:
@RingerJK , did you test with R8? I can help with testing if you haven't tested it yet

Reply:
there's no reflection that must be tested. We changed the approach from checking package name to checking interface

Reply:
I once had an issue when R8 merged all interfaces which object implements in one. I'm not sure that it will be the case for us, because user's router doesn't implement `InternalRouter` and probably R8 is clever enough to not to merge interfaces in our case, but I'm not sure. I'm not confident about this line. That's why I'm ready to help with testing.